Extension/src/LanguageServer/client.ts

```diff
@@ -980,8 +1101,22 @@ export class DefaultClient implements Client {
                         }
                         // Listen for messages from the language server.
                         this.registerNotifications();
-                    } else {
+                    }
+                    // update all client configurations
+                    this.configuration.setupConfigurations();
+                    initializedClientCount++;
+                    // count number of clients, once all clients are configured, check for trusted compiler to display notification to user and add a short delay to account for config provider logic to finish
+                    if ((vscode.workspace.workspaceFolders === undefined) || (initializedClientCount >= vscode.workspace.workspaceFolders.length)) {
```

Comment:
This code gets called again if the language server crashes. It seems like we wouldn't want this to be called again in that case, right?

Reply:
I'm seeing some infinite client creation occuring after a crash -- it looks like some existing code is creating a client in a deactivated state and it's invalid to make a cpptools request for that client since the cpptols process crashed, so it throws an exception.